src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_calculated_documents.py

```diff
@@ -0,0 +1,677 @@
+) -> Optional[CalculatedDocument]:
+    well_items = view_data.get_leaf_item(sample, target)
+    if (adj_eq_ct_mean := well_items[0].result.adj_eq_ct_mean) is None:
+        return None
+
+    eq_ct_mean_ref = build_eq_ct_mean(view_data, sample, target)
+    if eq_ct_mean_ref is None:
+        return None
+
+    return CalculatedDocument(
+        uuid=random_uuid_str(),
+        name="adjusted equivalent ct mean",
+        value=adj_eq_ct_mean,
+        data_sources=[
+            DataSource(feature="equivalent ct mean", reference=eq_ct_mean_ref)
+        ],
+    )
+
+
+@cache
+def build_delta_ct_mean(
+    view_data: ViewData[WellItem],
+    sample: str,
+    target: str,
+    r_target: str,
+) -> Optional[CalculatedDocument]:
+    well_items = view_data.get_leaf_item(sample, target)
+    if (delta_ct_mean := well_items[0].result.delta_ct_mean) is None:
+        return None
+
+    data_sources = []
+
+    if adj_eq_ct_mean_ref := build_adj_eq_ct_mean(view_data, sample, target):
+        data_sources.append(
+            DataSource(
+                feature="adjusted equivalent ct mean",
+                reference=adj_eq_ct_mean_ref,
+            )
+        )
+    elif eq_ct_mean_ref := build_eq_ct_mean(view_data, sample, target):
+        data_sources.append(
+            DataSource(
+                feature="equivalent ct mean",
+                reference=eq_ct_mean_ref,
+            )
+        )
+    else:
+        return None
+
+    if r_adj_eq_ct_mean_ref := build_adj_eq_ct_mean(view_data, sample, r_target):
+        data_sources.append(
+            DataSource(
+                feature="adjusted equivalent ct mean",
+                reference=r_adj_eq_ct_mean_ref,
+            )
+        )
+    elif r_eq_ct_mean_ref := build_eq_ct_mean(view_data, sample, r_target):
+        data_sources.append(
+            DataSource(
+                feature="equivalent ct mean",
+                reference=r_eq_ct_mean_ref,
+            )
+        )
+    else:
+        return None
+
+    return CalculatedDocument(
+        uuid=random_uuid_str(),
+        name="delta equivalent ct mean",
+        value=delta_ct_mean,
+        data_sources=data_sources,
+    )
+
+
+def build_delta_ct_sd(
+    view_data: ViewData[WellItem], sample: str, target: str, r_target: str
+) -> Optional[CalculatedDocument]:
+    well_items = view_data.get_leaf_item(sample, target)
+    if (delta_ct_sd := well_items[0].result.delta_ct_sd) is None:
+        return None
+
+    ct_sd_ref = build_ct_sd(view_data, sample, target)
+    if ct_sd_ref is None:
+        return None
+
+    r_ct_sd_ref = build_ct_sd(view_data, sample, r_target)
+    if r_ct_sd_ref is None:
+        return None
+
+    return CalculatedDocument(
+        uuid=random_uuid_str(),
+        name="delta equivalent ct sd",
+        value=delta_ct_sd,
+        data_sources=[
+            DataSource(feature="ct sd", reference=ct_sd_ref),
+            DataSource(feature="ct sd", reference=r_ct_sd_ref),
```

Comment:
Just confirming - we don't signify which value is reference value? I'm confused how you can tell from the data document which value is which.

Reply:
Yes, the feature is the same. The way that the reference is determined is with the data_processing_document when the calculated data documents are constructed in the schema.

Reply:
@nathan-stender @alejandro-salgado - It is hard to check all of these manually based on the JSON - but the `calculated data documents` and their associated `data source documents` look okay to me.